docs/csi-volume-expansion.md

```diff
@@ -0,0 +1,186 @@
+# Expanding CSI persistent volumes with Kubernetes
+
+## Introduction
+
+The Pure Service Orchestrator Kubernetes CSI driver includes support for CSI volume expansion. 
+
+The feature allows Kubernetes end-users to expand a persistent volume (PV) after creation by resizing a dynamically provisioned persistent volume claim (PVC).
+The end-users are able to edit the `allowVolumeExpansion` boolean flag in Kubernetes `StorageClass` to modify the permission whether a PVC resizing is allowed. 
+
+## Prerequisites
+
+The following prerequisites must be true to ensure the function can work properly:
+
+* Only dynamically provisioned PVCs can be resized.
+* Only allow volume size expansion, shrinking a volume is not allowed.
+* The `StorageClass` that provisions the PVC must support resize. `allowVolumeExpansion` flag is set to true by default in all PSO `StorageClass`.
+* The PVC `accessMode` must be `ReadWriteOnce` or `ReadWriteMany`.
+
+### Note
+
+Any PVC creted using a StorageClass where the `parameters.backend=block` will only be resized upon a (re)start of the pod bound to the PVC. 
+
+## Dependencies
+
+* Kubernetes already running, deployed, configured, etc.
+* Kubernetes supports CSI Volume Expansion as beta since 1.16 and GA since Kubernetes 1.19,
+  ensure you have Kubernetes 1.17+ (required by PSO  6.0+) installed.
+  [More Info](https://kubernetes-csi.github.io/docs/volume-expansion.html)
+* PSO 6.0+ is correctly installed.
+
+## Example usages
+
+PSO CSI driver supports `ONLINE` volume expansion capability, i.e. expanding an in-use PersistentVolumeClaim.
+For more details check the [CSI spec](https://github.com/container-storage-interface/spec/blob/master/spec.md) please. 
+
+**Note:** the expanded PVC requires a pod restart if you are using FlashArray as a backend, but not required for FlashBlade.
+
+### Before you run, ensure `allowVolumeExpansion` is set to `true` in your StorageClass:
+
+PSO StorageClass `pure-block` and `pure-block` have `allowVolumeExpansion` set to `true` by default.
```

Comment:
`pure-block` and `pure-block`?

Reply:
oh fixed.. hope this version might look cleaner.